### ISSUE TITLE
Allow reorganisation of favourites

### ISSUE THREAD

**Issue by UltraBlackLinux** [heqro/stupid-simple-launcher#1]
Hey there,
lovely launcher, but there is one thing I'd really like to see: I want to group specific favourites together so that I can find specific programs more easily, but I can't reorganise them.
Could you add that?
Thanks!

**Comment by Goldaracena** [comment]
Thank you for the comment. I will most likely get around that once I'm done with this university year, so this will definitely take some time. The reason for this is because (a) my knowledge of QML is this project's age old and (b) I'm using KDE's private API for receiving the user's applications. For doing those cool tricks (such as grouping and other stuff the guys at the [pling website](https://www.pling.com/p/1584342/) proposed me) I will have to parse whatever KDE's private API does.

Long story short, a myriad of suggestions I've been receiving _should_ be doable. But I will need some patience: the kind comments I've read are all written down in my to-do list, so I promise I will get around them!

**Comment by Tuxman2** [comment]
I will be great if we could reorganize the favorites by moving them in the favorites list. This is another suggestion. ;-)

**Comment by Y0Mamma** [comment]
This is a very much needed feature. As it is now, I have to remove things from my favorites and readd them in the order I want them on screen. Is there a way to do this in KDE outside of the menu until you get this feature added?
Oh, btw, this is the best menu out there, and I have tried them all. Do you accept bit coin donations?

**Comment by Tuxman2** [comment]
@Y0Mamma: We have to wait, Heqro is looking for to implement it. It takes time. You can reorganize your favorites via the desktop panel (edit panel) as you can add favorites on it.

**Comment by Goldaracena** [comment]
>I have to remove things from my favorites and readd them in the order I want them on screen.

Indeed, this is how it works as of now. I definitely want to implement it. I just am having a bit of trouble to find the time to do it right out of the gate.

>Is there a way to do this in KDE outside of the menu until you get this feature added?

I'm not too certain.

>Oh, btw, this is the best menu out there, and I have tried them all.

Thank you so much!

>Do you accept bit coin donations?

I will accept them once I think the menu is actually reliable and polished. There's still loads and loads of work for me to do. You guys finding bugs, proposing visual improvements and such things is more than enough for me now. This is my first project in qml and plasmoid development. I still have to find my mojo.

**Comment by Goldaracena** [comment]
This [feature has been added](https://github.com/heqro/stupid-simple-launcher/commit/c4b18585accd0a341f6f2f7563f97a7398ebd0f2) as of now and is currently present in the `develop` branch. Given that you guys seem to know how to use GitHub arguably better than me, if you want to try it I urge you to swap to this branch and download that branch's menu version.

The reason why change this hasn't already been pushed into `master` is because I'd like to keep track of behavior of the menu in my device for a while as KDE doesn't have any testing suite for plasmoids. This change was extremely simple as well, so that's another reason why I'm hesitant to push it immediately to `master`.

All in all, the next release for the menu is shaping up to look great. 

You guys tell me if it works alright.

**Comment by Tuxman2** [comment]
Ok, I download the version from the develop branch and I will give you my impression (probably for tomorrow).  ;-)

**Comment by Tuxman2** [comment]
Reorganisation of favourites tested. It works. :-)

**Comment by Goldaracena** [comment]
Okay, closing down this issue.
